qiskit_ibm_provider/transpiler/passes/scheduling/scheduler.py

```diff
@@ -0,0 +1,264 @@
+
+        Conditional execution durations are currently non-deterministic. as we do not know
+        the time it will take to begin executing the block. We do however know when the
+        duration of the block contents execution (provided it does not also contain
+        conditional executions).
+
+        TODO: Update for support of general control-flow, not just single conditional operations.
+        """
+        # Special processing required to resolve conditional scheduling dependencies
+        if node.op.condition_bits:
+            # Trigger the start of a conditional block
+            self._begin_new_circuit_block()
+
+            op_duration = self._get_duration(node)
+
+            t0q = max(self._idle_after[q][1] for q in node.qargs)
+            # conditional is bit tricky due to conditional_latency
+            t0c = max(self._idle_after[bit][1] for bit in node.op.condition_bits)
+            if t0q > t0c:
+                # This is situation something like below
+                #
+                #           |t0q
+                # Q ▒▒▒▒▒▒▒▒▒░░
+                # C ▒▒▒░░░░░░░░
+                #     |t0c
+                #
+                # In this case, you can insert readout access before tq0
+                #
+                #           |t0q
+                # Q ▒▒▒▒▒▒▒▒▒▒▒
+                # C ▒▒▒░░░▒▒░░░
+                #         |t0q
+                #
+                t0c = max(t0q, t0c)
+            t1c = t0c
+            for bit in node.op.condition_bits:
+                # Lock clbit until state is read
+                self._idle_after[bit] = (self._current_block_idx, t1c)
+
+            # It starts after register read access
+            t0 = max(t0q, t1c)  # pylint: disable=invalid-name
+
+            t1 = t0 + op_duration  # pylint: disable=invalid-name
+            self._update_idles(node, t0, t1)
+
+            # Terminate the conditional block
+            self._begin_new_circuit_block()
+
+        else:
+            # Fall through to generic case if not conditional
+            self._visit_generic(node)
+
+    def _visit_measure(self, node: DAGNode) -> None:
+        """Visit a measurement node.
+
+        Measurement currently triggers the end of a pulse block in IBM dynamic circuits hardware.
+        This means that it is possible to schedule *up to* a measurement (and during its pulses)
+        but the measurement will be followed by a period of indeterminism.
+        All measurements on disjoint qubits will be collected on the same qubits
+        to be run simultaneously."""
+        current_block_measure_qargs = self._current_block_measure_qargs()
+        measure_qargs = set(node.qargs)
+
+        t0q = max(
+            self._idle_after[q][1] for q in measure_qargs
+        )  # pylint: disable=invalid-name
+
+        # If the measurement qubits overlap, we need to start a new scheduling block.
+        if current_block_measure_qargs & measure_qargs:
```

Comment:
It this sufficient condition to terminate current block? i.e.
```python
qc = QuantumCircuit(2, 1)
qc.measure(0, 0)
qc.measure(0, 0)
qc.x(1)  # This will never come after the second measurement?
```
as far as I can check, `topological_op_nodes` always returns the x(1) node before the measurements. But I'm not sure how retworkx determines the node ordering. @mtreinish